qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/AlternativeRouteActivity.kt

```diff
@@ -120,21 +130,42 @@ class AlternativeRouteActivity : AppCompatActivity(), OnMapLongClickListener {
                 mapboxNavigation.unregisterRouteProgressObserver(replayProgressObserver)
                 mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
                 mapboxNavigation.unregisterLocationObserver(locationObserver)
+                mapboxNavigation.unregisterRoutesObserver(recordRoutesObserver)
                 mapboxNavigation.unregisterRoutesObserver(routesObserver)
                 mapboxNavigation.unregisterRouteAlternativesObserver(alternativesObserver)
             }
         },
         onInitialize = this::initNavigation
     )
 
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        LogConfiguration.setLoggingLevel(LoggingLevel.DEBUG)
         setContentView(binding.root)
-
         initStyle()
         initListeners()
     }
 
+    @OptIn(ExperimentalMapboxNavigationAPI::class)
+    private val fasterRoutes by lazy {
+        mapboxNavigation.createFasterRoutes(
+            FasterRouteOptions.Builder().build()
+        )
+    }
+
+    override fun onResume() {
+        super.onResume()
+        @OptIn(ExperimentalMapboxNavigationAPI::class)
+        fasterRoutes.registerNewFasterRouteObserver(fasterRouteObserver)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        @OptIn(ExperimentalMapboxNavigationAPI::class)
+        fasterRoutes.unregisterNewFasterRouteObserver(fasterRouteObserver)
+    }
+
```

Comment:
I'm sure there's a better way to create `FasterRoute` and keep it as long as `MapboxNavigation`. Any suggestions?

Reply:
Found it -`MapboxNavigationObserver`